sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -84,7 +84,6 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
       job: Job,
       options: Map[String, String],
       dataSchema: StructType): OutputWriterFactory = {
-    verifySchema(dataSchema)
```

Comment:
Actually, I was hesitated to change this because It now throws an exception from executor-side but I just decided to unify them just to clean up.

I can do this as it was but adding more unsupported types if you think this is more sensible.